security/pkg/pki/cainterface/ca.go

```diff
@@ -0,0 +1,32 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cainterface
+
+import (
+	"time"
+
+	"istio.io/istio/security/pkg/pki/util"
+)
+
+// CertificateAuthority contains methods to be supported by a CA.
+type CertificateAuthority interface {
```

Comment:
Please don't create a file for the interface. This is Java pattern, not Golang pattern. In Golang, you define the interface when you use it.
https://blog.chewxy.com/2018/03/18/golang-interfaces/